pkg/azuredevops/smoke/submission.go

```diff
@@ -0,0 +1,511 @@
+	state  string
+	result string
+}
+
+func main() {
+	// Fetching environment variables for Azure DevOps settings
+	organizationUrl := os.Getenv("ORGANIZATION_URL")
+	personalAccessToken := os.Getenv("PERSONAL_ACCESS_TOKEN")
+	projectName := os.Getenv("PROJECT_NAME")
+	pipelineName := os.Getenv("PIPELINE_NAME")
+	pipelineIdStr := os.Getenv("PIPELINE_ID")
+	buildIdStr := os.Getenv("BUILD_ID")
+
+	// Converting variables from string to integer
+	pipelineId, err := strconv.Atoi(pipelineIdStr)
+	if err != nil {
+		log.Fatalf("Error parsing PIPELINE_ID: %v", err)
+	}
+	buildId, err := strconv.Atoi(buildIdStr)
+	if err != nil {
+		log.Fatalf("Error parsing BUILD_ID: %v", err)
+	}
+
+	// Setting up context for API calls
+	ctx := context.Background()
+
+	// Creating a connection to Azure DevOps using the Personal Access Token
+	connection := createPatConnection(organizationUrl, personalAccessToken)
+
+	// Determining which tests to run based on the TESTS_TO_RUN environment variable
+	testsToRun := os.Getenv("TESTS_TO_RUN")
+
+	var testsToRunList []string
+	if testsToRun != "" && testsToRun != "all" {
+		for _, test := range strings.Split(testsToRun, ",") {
+			testsToRunList = append(testsToRunList, strings.TrimSpace(test))
+		}
+	}
+
+	// Defining build tests with their descriptions and expected log messages
+	buildTests := []buildTest{
+		{
+			description:  "Checkout self repository",
+			logMessage:   "Repository 'self' has been successfully checked out.",
+			expectAbsent: false,
+		},
+		{
+			description:  "Checkout kyma-modules repository",
+			logMessage:   "Repository 'kyma-modules' has been successfully checked out.",
+			expectAbsent: false,
+		},
+		{
+			description:  "Checkout security-scans-modular repository",
+			logMessage:   "Repository 'security-scans-modular' has been successfully checked out.",
+			expectAbsent: false,
+		},
+		{
+			description:  "Verify the download of conduit-cli",
+			logMessage:   "conduit-cli has been successfully downloaded and is executable.",
+			expectAbsent: false,
+		},
+		{
+			description:  "Verify Python Installation",
+			logMessage:   "command not found: python3",
+			expectAbsent: true,
+		},
+		{
+			description:  "Verify gcloud Installation",
+			logMessage:   "command not found: gcloud",
+			expectAbsent: true,
+		},
+		{
+			description:  "Verify Go Installation",
+			logMessage:   "command not found: go",
+			expectAbsent: true,
+		},
+		{
+			description:  "Download Kyma cli",
+			logMessage:   "The 'kyma' binary has been successfully downloaded and is executable.",
+			expectAbsent: false,
+		},
+	}
+
+	// Running each build test if it meets the criteria specified in TESTS_TO_RUN
+	for _, test := range buildTests {
+		if shouldRunTest(testsToRun, testsToRunList, test.description) {
+			runBuildTest(ctx, connection, projectName, pipelineName, pipelineId, test)
+		}
+	}
+
+	// Defining timeline tests with their names and expected states and results
+	timelineTests := []timelineTest{
```

Comment:
Do you think this file can be refactored into multiple ones e.g for different types of tests? I think this file is quite long and we have mixed responsibilities here, we have tests configuration, logic for execution and checkBuildRecords helper functions which should be in different pkg file and only used in the cmd file.

Reply:
Hmm, yes this can be for sure reworked for cmd/pkg.